### PR TITLE
Adding --set-string flag to force string values.

### PR DESCRIPTION
**Pull request by arturo-c**
This addresses the problem of not being able to pass a long int value via the --set flag, and helm converting it to float64 scientific notation, one can then manipulate the string as needed inside the helm templates:

Example usage:
`helm install --set long_string_value=1234567890` results in value once in helm template of: `1.23456789e+09`

with new flag:
`helm install --set-string long_string_value=1234567890` -> value = `"1234567890"`

Addresses these github issues:
https://github.com/kubernetes/helm/issues/1707
https://github.com/kubernetes/helm/issues/1694
https://github.com/kubernetes/helm/issues/3155
https://github.com/kubernetes/helm/issues/3001
https://github.com/kubernetes/helm/issues/3246

This is my first dive at go, so please be gentle if I'm doing something completely wrong :)

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@arturo-c Thanks for the contribution! I will take a look at this on Monday. If for some reason I don't get around to it, feel free to ping me. Sorry for the delay on this

File: pkg/strvals/parser.go
Comment (thomastaylor312) [inline pkg/strvals/parser.go]: I think it may be clearer and won't break the current API if you add a new method called `ParseIntoString`. The other private methods can still use a `bool`, but for someone consuming this package, they won't need to change how they are calling the methods. There might even be a better idea from one of the other maintainers as well. `ParseIntoString` seems pretty simple though

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
One change to make sure we don't break publicly exposed APIs, otherwise this looks pretty straightforward and you added unit tests 🎉

**Comment by arturo-c** [comment]
@thomastaylor312 thanks for the input, I had to add two public methods `Parse` and `ParseIntoString`, definitely will keep in mind not changing publicly exposed methods next time 👍

File: cmd/helm/template.go
Comment (thomastaylor312) [inline cmd/helm/template.go]: Looks like a copy/paste error. This should be `t.stringValues`.

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Sorry, one more thing I missed in initial review that I caught during some testing. After that, this should be good to go with one more review

File: cmd/helm/template.go
Comment (arturo-c) [reply cmd/helm/template.go]: Nice catch, updated 👍

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
@arturo-c It looks like this should be rebased against master because it is overwriting changes on helm/master

**Comment by arturo-c** [comment]
@thomastaylor312 cool, not sure what happened there, but fixed the commit.

**Comment by thomastaylor312** [comment]
@arturo-c So because this is labeled as a large, it will need 2 reviews before merge. We have the dev call tomorrow, so I will make sure someone reviews it. Great work on this!

**Review by bacongobbler — APPROVED** [review]
this looks like a good approach. Thanks!

**Comment by babatundebusari** [comment]
@arturo-c 
here is what i have when i have --set-string for my helm install command

`Error: unknown flag: --set-string`

**Comment by bacongobbler** [comment]
you can test it with either the [canary releases](https://docs.helm.sh/using_helm/#from-canary-builds) or [v2.9.0-rc4](https://github.com/kubernetes/helm/releases/tag/v2.9.0-rc4) :)

**Comment by OndraZizka** [comment]
While I consider this a workaround for the bug #1707 , it will be useful until the fix for it is released.

However. Will this be also applied to `helm lint`? See #4098

**Comment by OndraZizka** [comment]
Btw when is this supposed to be released? I checked with 2.9.1 and gett this:
```
+ helm install --replace --name h2dc6 --wait --timeout 600 ./csa-application/src/main/helm/charts/collab-services-api --set-string image.tag=2dc667a5
Error: unknown flag: --set-string
```

**Comment by ashahba** [comment]
@babatundebusari and @OndraZizka What is your: `helm version` output.
If you don't get:
```
$ helm version
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```
Then I recommend first updating your client version to `v2.9.0` and then run `helm init --upgrade` so both client and server run matching versions.
Finally you can use `--set-string` option.

**Comment by OndraZizka** [comment]
I think both server and client were 2.9.1 in my case.

**Comment by OndraZizka** [comment]
Anyway - does the introduction of `--set-string` mean that `--set` sets other type? What type? Is it a generic number? Or can we expect more surprises, like, "516F" will be converted to Celsius?

**Pull request merged**
